### PR TITLE
Patch fs2 blobstore and s3 store with fs2 1.0.0-M4

### PR DESCRIPTION
**Pull request by Igosuki**
I haven't patched box and sftp but it shouldn't be hard, would you like me to finish ?

### CODE REVIEW

File: core/src/main/scala/blobstore/StoreOps.scala [hunk 1]
```diff
@@ -17,15 +17,15 @@ package blobstore
...
+import cats.effect.{ConcurrentEffect, ContextShift, Effect, Sync}
 import fs2.{Pipe, Sink, Stream}
 import cats.implicits._
 
 import scala.concurrent.ExecutionContext
 
 trait StoreOps {
 
-  implicit class PutOps[F[_]](store: Store[F]) {
+  implicit class PutOps[F[_] : ContextShift](store: Store[F])(implicit ec: ExecutionContext) {
```
Comment (Daenyth) [inline core/src/main/scala/blobstore/StoreOps.scala]: Is it possible to move the constraints to the functions themselves? This makes the class more general

File: core/src/test/scala/blobstore/AbstractStoreTest.scala [hunk 2]
```diff
@@ -177,7 +182,7 @@ trait AbstractStoreTest extends FlatSpec with MustMatchers with BeforeAndAfterAl
       c1 must be(c2)
     }
 
-    test.unsafeRunSync()
+    test
```
Comment (Daenyth) [inline core/src/test/scala/blobstore/AbstractStoreTest.scala]: Double check this is running - `FlatSpec` tests want you to return `Any`, so you won't be warned of a discarded `IO`. Based on my reading I think this test is not being executed

File: s3/src/main/scala/blobstore/s3/S3Store.scala [hunk 3]
```diff
@@ -18,39 +18,39 @@ package s3
...
 
 import scala.concurrent.ExecutionContext
 
-case class S3Store[F[_]](s3: AmazonS3, sse: Option[String] = None)(implicit F: Effect[F], ec: ExecutionContext) extends Store[F] {
+case class S3Store[F[_] : ConcurrentEffect : ContextShift](s3: AmazonS3, sse: Option[String] = None)(implicit F: Effect[F], ec: ExecutionContext) extends Store[F] {
 
-  // convert ObjectListing to Segment[Path, Unit]
-  private def _segment(ol: ObjectListing): Segment[Path, Unit] = {
-    val dirs: Segment[Path, Unit] = Segment.seq(ol.getCommonPrefixes.asScala.map(s =>
+  // convert ObjectListing to Chunk[Path]
```
Comment (Daenyth) [inline s3/src/main/scala/blobstore/s3/S3Store.scala]: Not yours, but IMO this comment can go - it repeats the line below

File: s3/src/main/scala/blobstore/s3/S3Store.scala [hunk 4]
```diff
@@ -18,39 +18,39 @@ package s3
...
+  private def _segment(ol: ObjectListing): Chunk[Path] = {
```
Comment (Daenyth) [inline s3/src/main/scala/blobstore/s3/S3Store.scala]: Rename the function too?

File: s3/src/main/scala/blobstore/s3/S3Store.scala [hunk 5]
```diff
@@ -65,7 +65,7 @@ case class S3Store[F[_]](s3: AmazonS3, sse: Option[String] = None)(implicit F: E
 
   override def get(path: Path, chunkSize: Int): Stream[F, Byte] = {
     val is: F[InputStream] = F.delay(s3.getObject(path.root, path.key).getObjectContent)
-    fs2.io.readInputStream(is, chunkSize, closeAfterUse = true)
+    fs2.io.readInputStream(is, chunkSize, closeAfterUse = true, blockingExecutionContext = ec)
```
Comment (Daenyth) [inline s3/src/main/scala/blobstore/s3/S3Store.scala]: IMO if it's going to be used as a blocking pool, I'd make the implicit `ec` on the class constructor explicit and name it to match this one.

**Comment by Igosuki** [comment]
Thanks, I'll have a look.

Le mar. 11 sept. 2018 à 18:37, Gavin Bisesi <notifications@github.com> a
écrit :

**Comment by rolandomanrique** [comment]
@Igosuki thanks for the PR, finishing sftp and box would be great too.

**Comment by Igosuki** [comment]
I merged the PR, thanks for the updates as I didn't have the time to finish it at all

**Comment by rolandomanrique** [comment]
@Daenyth @Igosuki thanks. I see travis is failing I will review and run tests locally and let you know.

**Comment by Daenyth** [comment]
It looks like the tests hung - I should have run them locally, forgot to since I was working on them late at night :P

I'd immediately be suspicious of places where we use `blockingExecutionContext = <TestContext>` in the tests, try switching that to `global`

**Comment by Daenyth** [comment]
The way I used TestContext is wrong; I'll fix it

**Comment by Daenyth** [comment]
https://github.com/Adikteev/fs2-blobstore/pull/2 @Igosuki

**Comment by Igosuki** [comment]
Done

Le mar. 25 sept. 2018 à 14:43, Gavin Bisesi <notifications@github.com> a
écrit :

**Pull request merged**